src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,124 @@
+import React, {
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type TodoItemProps = {
+  todo: Todo;
+  isLoading: boolean;
+  onDelete: (id: number) => void;
+  onStatusChanged: () => void;
+  updateTodo: (id: number, newTitle: string) => void;
+};
+
+export const TodoItem: React.FC<TodoItemProps> = ({
+  todo: { id, title, completed },
+  isLoading,
+  onDelete,
+  onStatusChanged,
+  updateTodo,
+}) => {
+  const [editMode, setEditMode] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const handleTitleUpdate = useCallback(() => {
+    if (!newTitle.trim()) {
+      onDelete(id);
+    } else if (newTitle !== title) {
+      updateTodo(id, newTitle);
+    }
+
+    setEditMode(false);
+  }, [newTitle]);
+
+  const onFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    handleTitleUpdate();
+  };
+
+  const inputTitle = useRef<HTMLInputElement>(null);
+
+  const handleClose = () => {
+    setNewTitle(title);
+    setEditMode(false);
+  };
+
+  useEffect(() => {
+    if (inputTitle.current !== null) {
+      inputTitle.current.focus();
+    }
+
+    const handleEscape = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        handleClose();
+      }
+    };
+
+    document.addEventListener('keydown', handleEscape);
+
+    return () => {
+      document.removeEventListener('keydown', handleEscape);
+    };
+  }, [editMode]);
```

Comment:
The `useEffect` function is missing the necessary dependency `handleClose` in the dependency array.